doc/design-component.md

```diff
@@ -0,0 +1,69 @@
+Component Loader Design
+=======================
+
+# Background
+MicroShift is designed to provide a turnkey solution that can readily serve the target workloads 
+with little intervention, probably in an airgapped environment. This requires MicroShift to
+be able to provides the most, if not all, infrastructure components during bootstrapping. 
```

Comment:
"At the same time, one of MicroShift's design goals is to keep its fixed core functionality to a minimum while allowing to easily extend or swap out infrastructure components to match the use case."